staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -133,3 +136,133 @@ func TestWatchErrResultNotBlockAfterCancel(t *testing.T) {
 	cancel()
 	wg.Wait()
 }
+
+func TestWatchChanSync(t *testing.T) {
```

Comment:
What do you think about refactoring this test into a table test? 
Additionally, what do you think about splitting the test into 2 test cases? 
If we already have that, we could also consider adding new test cases for RV > 0.